src/lib/utils/string_utils.cpp

```diff
@@ -57,4 +60,17 @@ std::string replace_addresses(const std::string& input) {
   return std::regex_replace(input, std::regex{"0x[0-9A-Fa-f]{4,}"}, "0x00000000");
 }
 
+template <typename String>
+pmr_string string_to_lower(const String& input) {
+  auto copy = pmr_string{input};
```

Comment:
Instead of iterating the vector twice (first copy, then adapt) we could also `std::string::reserve` and then append the lower case to the vector.